File: Backend/src/models/boilerplate/sos.model.js
const mongoose = require('mongoose');
const { required } = require('joi');
const { toJSON, paginate } = require('../plugins');

const sosSchema = mongoose.Schema(
  {
    title: {
      type: String,
      trim: true,
      default: null,
    },
    phoneNumber: {
      type: String,
      trim: true,
      default: null,
    },
    countryCode: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Country',
    },
    dialCode: {
      type: String,
      trim: true,
      default: null,
    },
    status: {
      type: Boolean,
      default: true,
    },
    deletedAt: {
      type: Date,
      default: null,
    },
    isAdminAdded: {
      type: Boolean,
      default: false,
    },
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Client',
    },
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: false,
      required: false,
    },

    phoneNumber: {
      type: String,
      trim: true,
      default: null,
    },
    status: {
      type: Boolean,
      default: true,
    },
    deletedAt: {
      type: Date,
      default: null,
    },
    isAdminAdded: {
      type: Boolean,
      default: false,
    },
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Client',
    },
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: false,
      required: false,
    },
  },
  {
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
    },
  },
);

// Add plugin that converts mongoose to json
sosSchema.plugin(toJSON);
sosSchema.plugin(paginate);

/**
 * @typedef SOS
 */
const SOS = mongoose.model('SOS', sosSchema);

module.exports = SOS;

File: User/lib/network/response_models/sos_model.dart
class SOSModel {
  String? dialCode;
  String? title;
  String? phoneNumber;
  bool? status;
  Null? deletedAt;
  bool? isAdminAdded;
  String? clientId;
  String? userId;
  String? id;
  String? countryCode;

  SOSModel(
      {this.dialCode,
        this.title,
        this.phoneNumber,
        this.status,
        this.deletedAt,
        this.isAdminAdded,
        this.clientId,
        this.userId,
        this.id,
        this.countryCode});

  SOSModel.fromJson(Map<String, dynamic> json) {
    dialCode = json['dialCode'];
    title = json['title'];
    phoneNumber = json['phoneNumber'];
    status = json['status'];
    deletedAt = json['deletedAt'];
    isAdminAdded = json['isAdminAdded'];
    clientId = json['clientId'];
    userId = json['userId'];
    id = json['id'];
    countryCode = json['countryCode'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['dialCode'] = this.dialCode;
    data['title'] = this.title;
    data['phoneNumber'] = this.phoneNumber;
    data['status'] = this.status;
    data['deletedAt'] = this.deletedAt;
    data['isAdminAdded'] = this.isAdminAdded;
    data['clientId'] = this.clientId;
    data['userId'] = this.userId;
    data['id'] = this.id;
    data['countryCode'] = this.countryCode;
    return data;
  }
}

